model/approval_validate.go

```diff
@@ -33,20 +33,8 @@ func validateApprovals(approvals []*ApprovalPolicy) error {
 		return errors.New("No approval policies specified")
 	}
 	tailScope := approvals[len(approvals)-1].Scope
-	if len(tailScope.Paths) > 0 {
-		errs = multierror.Append(errs, errors.New("Final scope must have no paths"))
-	}
-	if len(tailScope.Branches) > 0 {
-		errs = multierror.Append(errs, errors.New("Final scope must have no branches"))
-	}
-	if len(tailScope.PathRegexp) > 0 {
-		errs = multierror.Append(errs, errors.New("Final scope must have no path regexp"))
-	}
-	if len(tailScope.BaseRegexp) > 0 {
-		errs = multierror.Append(errs, errors.New("Final scope must have no base branch regular expressions"))
-	}
-	if len(tailScope.CompareRegexp) > 0 {
-		errs = multierror.Append(errs, errors.New("Final scope must have no compare branch regular expressions"))
+	for _, m := range tailScope.NonFinal() {
+		errs = multierror.Append(errs, errors.New(m))
```

Comment:
Should NonFinal() return []error?

And as a maybe-related question: do we want to switch over to the new multierror package that I created? That's a far bigger effort.

Reply:
To be clear, I don't mean that we should switch multierrors in this ticket.